src/data/calendar.js

```diff
@@ -0,0 +1,139 @@
+    var firstOfMonth = utcDate(year, month, 1);
+    var lastOfMonth = utcDate(year, month + 1, 0);
+    var used = firstOfMonth.getUTCDay() + 6 + lastOfMonth.getDate();
+    var numberOfWeeks = Math.ceil(used / 7);
+
+    var weeks = [];
+    for (var i = 0; i < numberOfWeeks; i++) {
+      weeks.push([{}, {}, {}, {}, {}, {}, {}]);
+    }
+    var firstWeekDayOfMonth = (firstOfMonth.getUTCDay() - 1 + 7) % 7;
+
+    var frenchBankHolidays = getFrenchBankHolidays(year)
+    var bankHolidays = frenchBankHolidays
+        .map((holiday) => formatDateString(holiday.date));
+    var boxingDays = getBoxingDays(year, frenchBankHolidays)
+        .map((holiday) => formatDateString(holiday.date));
+
+    if (year < 2017) boxingDays = [];
+
+    return weeks.map((week, weekIndex) => {
+      return week.map((day, dayIndex) => {
+        var globalIndex = weekIndex * 7 + dayIndex;
+        var dayOfMonth = globalIndex - firstWeekDayOfMonth + 1;
+        day.date = utcDate(year, month, dayOfMonth);
+
+        if (day.date.getUTCMonth() === month - 1) {
+          var currentDate = formatDateString(day.date);
+          day.hours = 8;
+          if ([0, 6].includes(day.date.getUTCDay())) {
+            day.hours = 8;
+            day.type = 'WE';
+          } else if (bankHolidays.includes(currentDate)) {
+            day.type = 'JF';
+          } else if (boxingDays.includes(currentDate)) {
+            day.type = 'CS';
+          } else {
+            day.type = 'JT';
+          }
+        } else {
+          day.type = null;
+        }
+        return day;
+      });
+    });
+  }
+
+  function isInCurrentMonth(startDate, endDate, month, year) {
+    return startDate.getUTCMonth() <= month - 1 &&
+           startDate.getFullYear() <= year &&
+           endDate.getUTCMonth() >= month - 1 &&
+           endDate.getFullYear() >= year;
+  }
+
+  function updateWeeksWithHolidays(state) {
+    // Update the weeks data with the PTO infos for the current month.
+    state.holidays.forEach(({start, end, hours, comment}) => {
+      const startDate = new Date(start);
+      const endDate = new Date(end);
+
+      // Handle case where holidays ends on a week-end
+      while ([0, 6].includes(endDate.getUTCDay())) {
+        endDate.setDate(endDate.getDate() - 1);
+      }
+
+      // Is in current month?
+      if (isInCurrentMonth(startDate, endDate, state.currentMonth, state.currentYear)) {
+        // Someday of this holiday are in the current month
+        var type = guessTypeFromComment(comment);
+
+        state.weeks.forEach(week => {
+          week.forEach(day => {
+            if (startDate <= day.date && endDate >= day.date) {
+              var isLast = endDate.toJSON() === day.date.toJSON();
+              if (day.type !== 'WE') {
+                if (day.type == 'JT') {
+                  day.type = type;
+                  // If there is still more than 8 hours, it is probably a full day off
+                  if (hours > 8) {
```

Comment:
wondering if we want `>= 8` here ?

I admit I still don't understand properly this algorithm.